_posts/2019-03-19-processing-a-csv-file-in-batch-with-sidekiq.markdown

```diff
@@ -0,0 +1,87 @@
+---
+layout: post
+title:  "Processing a CSV file in batch with Sidekiq"
+date: 2019-03-19 10:00:00
+categories: ["sidekiq"]
+author: "cleiviane"
+---
+
+[Sidekiq Pro](https://sidekiq.org/products/pro.html) comes with a great feature to process a collection of jobs as a batch, allowing them to be monitored as a group and executing a callback function when all the jobs are finished. This is useful when you need to load a lot of spreadsheet files into your database.  
+
+Recently, that was the case of one of [Ombu Labs](http://ombulabs.com)' clients. They needed to upload a CSV file with over 10 thousand rows of loans data, which makes processing the file synchronously impossible because the browser will time out after a few seconds. Breaking the file into smaller ones wasn't a good idea either, because it would take an unacceptable amount of time to finish. So we decided to use the Sidekiq's batch logic.
+
+Since the client didn't want to spend money paying for Sidekiq Pro, we had the challenge of implementing the same pattern that Sidekiq Pro uses in their Batches processing. This article will show how we did it.
+
+<!--more-->
+
+## Show me the code
+
+After the user uploads the CSV file we save the data into the database and schedule one job that will schedule one background job for each row into the saved CSV file. This is necessary because the uploaded files are too big and scheduling all the jobs in execution time would also time out.
```

Comment:
This is necessary because the uploaded files are too big, and trying to process them as a single file would eventually time out as well